### PR TITLE
Change release process

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,39 +1,51 @@
...
 
-1. Make sure the new version is present at `pkg/upgrade/versions.go`
 
-1. Prepare a changelog since last release. Get the `OAUTH_TOKEN` from (Github)[https://github.com/settings/tokens/new?description=GitHub%20Changelog%20Generator%20token] and select `repo:status` scope.
+1. Change the `versions.txt `so that it lists the target version of the Jaeger (if it is required). Don't touch the operator version it will be changed automatrically in the next ste.
 
-    ```
-    OAUTH_TOKEN=... make changelog
-    ```
+2. Run `make prepare-release VERSION=1.25.0`, using the operator version that will be released.
```
Comment (pavolloffay) [inline RELEASE.md:8]: nit: use only `1.` as the counter for the list

File: RELEASE.md [hunk 2]
```diff
@@ -1,39 +1,51 @@
...
    git commit -sm "Preparing release v1.25.0"
    ```
 
-1. Tag and push
+5. Once the changes above are merged and available in `master` tag it with the desired version, prefixed with `v`, eg. `v1.25.0`
 
     ```
-    git checkout master ## it's only possible to release from master for now!
-    git tag release/v1.25.0
-    git push git@github.com:jaegertracing/jaeger-operator.git release/v1.25.0
```
Comment (pavolloffay) [inline RELEASE.md:24]: any reason why the tag name is changed?

File: RELEASE.md [hunk 3]
```diff
@@ -1,39 +1,51 @@
...
+    git checkout master
```
Comment (pavolloffay) [inline RELEASE.md:21]: keep the previous comment about master if it is true

File: RELEASE.md [hunk 4]
```diff
@@ -1,39 +1,51 @@
...
 
-1. Apply generated OLM catalog files to [operatorhub.io](https://operatorhub.io)
+7. After the release, PRs needs to be created against the Operator Hub Community Operators repositories:
+
+    * One for the `upstream-community-operators`, used by OLM on Kubernetes. (https://github.com/k8s-operatorhub/community-operators)
+    * One for the `community-operators` used by OpenShift. (https://github.com/redhat-openshift-ecosystem/community-operators-prod)
+
+This can be done with the following steps:
+- Update master `git pull git@github.com:jaegertracing/jaeger-operator.git master`
+- Clone both repositories `upstream-community-operators` and `community-operators`
```
Comment (pavolloffay) [inline RELEASE.md:39]: add link to these repositories

Comment (rubenvp8510) [reply RELEASE.md:39]: The links are in the above paragraph, Do you think we need to add it here too?

File: RELEASE.md (hunk 2 above)
Comment (rubenvp8510) [reply RELEASE.md:24]: The reason is this:
 - In the previous process for trigger a release we push a tag with `release/v1.25.0`, this triggers the process , the old process creates the manifest and push it to master and creates the `v1.25.0 `tag
 - In the new process I proposed here, you send a PR with the `CHANGELOG.md`, the new manifest and the `versions.txt` changes. when the PR is merged you only need to push a tag it with `v1.25.0 `. I'm skipping one step, the step of tagging with `release/v1.25.0`

Does that makes sense?

Comment (rubenvp8510) [reply RELEASE.md:24]: This line changes the `versions.txt `: https://github.com/jaegertracing/jaeger-operator/pull/1535/files#diff-84879d57aeb7ca56c4e920fb01240d818d775283ebbcc2860c347d4507115a4cL39

I can do that, but involves more steps, more steps more prone to error.  (specially more automatic things that could fail) IMHO.

Comment (rubenvp8510) [reply RELEASE.md:24]: WDYT? @jpkrohling  @pavolloffay

File: RELEASE.md [hunk 5]
```diff
@@ -1,39 +1,51 @@
 # Releasing the Jaeger Operator for Kubernetes
 
-1. Update Jaeger version in `versions.txt`
+Steps to release a new version of the Jaeger Operator:
 
-1. Make sure the new version is present at `pkg/upgrade/versions.go`
 
-1. Prepare a changelog since last release. Get the `OAUTH_TOKEN` from (Github)[https://github.com/settings/tokens/new?description=GitHub%20Changelog%20Generator%20token] and select `repo:status` scope.
+1. Change the `versions.txt `so that it lists the target version of the Jaeger (if it is required). Don't touch the operator version it will be changed automatrically in the next ste.
```
[the remark this answers is not in the document]
Comment (rubenvp8510) [reply RELEASE.md:6]: Ups, thanks

File: RELEASE.md (hunk 2 above)
Comment (jpkrohling) [reply RELEASE.md:24]: Agree with @pavolloffay that the release process should instead just open a PR with the changes. A tag can then be created based on that. So, we are back at the `release/v.X.Y.Z` to trigger the release and `vX.Y.Z` with the release itself.

File: RELEASE.md [hunk 6]
```diff
@@ -1,39 +1,51 @@
...
 
     ```
     git pull git@github.com:jaegertracing/jaeger-operator.git master
     ```
 
-1. Apply generated OLM catalog files to [operatorhub.io](https://operatorhub.io)
+7. After the release, PRs needs to be created against the Operator Hub Community Operators repositories:
+
+    * One for the `upstream-community-operators`, used by OLM on Kubernetes. (https://github.com/k8s-operatorhub/community-operators)
+    * One for the `community-operators` used by OpenShift. (https://github.com/redhat-openshift-ecosystem/community-operators-prod)
```
Comment (jpkrohling) [inline RELEASE.md:35]: This should be a proper link. Like:

```suggestion
    * One for the [`community-operators`](https://github.com/redhat-openshift-ecosystem/community-operators-prod) used by OpenShift.
```

File: RELEASE.md (hunk 2 above)
Comment (jpkrohling) [reply RELEASE.md:24]: We talked after I left my previous comment, and I agree with the simplified approach: run `make prepare-release VERSION=1.25.0`, open the PR, merge, then tag with `v1.25.0` directly, without the intermediary tag.

File: RELEASE.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (pavolloffay) [reply RELEASE.md:6]: `ste` still hasn't been updated to `step`

File: RELEASE.md [hunk 7]
```diff
@@ -1,39 +1,51 @@
...
-    git checkout master ## it's only possible to release from master for now!
-    git tag release/v1.25.0
-    git push git@github.com:jaegertracing/jaeger-operator.git release/v1.25.0
+    git checkout master
+    git tag v1.25.0
+    git push git@github.com:jaegertracing/jaeger-operator.git v1.25.0
     ```
 
-1. Wait until release CI job finishes and then pull the changes:
+1. The GitHub Workflow will take it from here, creating a GitHub release and publishing the images
```
Comment (pavolloffay) [inline RELEASE.md:26]: the following git pull can be removed. It's documented later when updating the operator hub

**Review by pavolloffay — APPROVED** [review]

**Comment by rubenvp8510** [comment]
This was approved, can we merge it?

**Pull request merged**